libopae/init_ase.c

```diff
@@ -60,9 +62,45 @@ STATIC const char *_opae_sys_cfg_files[SYS_CFG_PATHS] = {
 	"/etc/opae/opae_ase.cfg",
 };
 
-// Find the canonicalized configuration file. If null, the file was not found.
-// Otherwise, it's the first configuration file found from a list of possible
-// paths. Note: The char * returned is allocated here, caller must free.
+/* global loglevel */
+static int g_loglevel = OPAE_DEFAULT_LOGLEVEL;
+static FILE *g_logfile;
+/* mutex to protect against garbled log output */
+static pthread_mutex_t log_lock = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
+
+void /* __FIXME_MAKE_VISIBLE__ */ opae_print(int loglevel, const char *fmt, ...)
+{
+	FILE *fp;
+	int err;
+	va_list argp;
+
+	if (loglevel > g_loglevel)
+		return;
+
+	if (loglevel == OPAE_LOG_ERROR)
+		fp = stderr;
+	else
+		fp = g_logfile == NULL ? stdout : g_logfile;
+
+	va_start(argp, fmt);
+	err = pthread_mutex_lock(
+		&log_lock); /* ignore failure and print anyway */
+	if (err)
+		fprintf(stderr, "pthread_mutex_lock() failed: %s",
+			strerror(err));
+	vfprintf(fp, fmt, argp);
+	err = pthread_mutex_unlock(&log_lock);
+	if (err)
+		fprintf(stderr, "pthread_mutex_unlock() failed: %s",
+			strerror(err));
+	va_end(argp);
+}
+
+/* Find the canonicalized configuration file opae_ase.cfg. If null, the file
+	was not found. Otherwise, it's the first configuration file found from a
```

Comment:
Indent is off (x3 lines)

Reply:
fixed

Reply:
align these 3 lines with the F in Find from 99.